src/Raven.Server/ServerWide/Commands/CompareExchangeCommands.cs

```diff
@@ -395,5 +421,20 @@ private static void ThrowCompareExchangeKeyTooBig(string str)
                 $"Compare Exchange key cannot exceed {MaxNumberOfCompareExchangeKeyBytes} bytes, " +
                 $"but the key was {Encoding.GetByteCount(str)} bytes. The invalid key is '{str}'. Parameter '{nameof(str)}'");
         }
+        
+        protected override bool Validate(ClusterOperationContext context, Slice keySlice, Table items, long index, out long currentIndex)
+        {
+            BlittableJsonReaderObject value;
+            (currentIndex, value) = ClusterStateMachine.GetCompareExchangeValue(context, keySlice, items);
+            if (currentIndex == InvalidIndexValue)
+                return index == 0;
+
+            using (value)
+            {
+                if (CompareExchangeExpirationStorage.TryGetExpires(value, out var ticks) && ticks < CurrentTicks)
```

Comment:
shouldn't we re-throw like we do in `protected bool TryGetExpires(BlittableJsonReaderObject value, string storageKey, out long ticks)`?